components/CatModal/CatModalHeader.tsx

```diff
@@ -20,91 +21,144 @@ export default function CatModalHeader({
   sterilizeCat: () => void;
 }) {
   const { t } = useTranslation('common');
+  const [confirmationVisible, setConfirmationVisible] = useState<boolean>(false);
+  const [deletePressed, setDeletePressed] = useState<boolean>(false);
   return (
-    <Group
-      p='md'
-      position='apart'
-      align='center'
-      sx={{
-        width: '100%',
-        borderRadius: '15px',
-      }}>
-      <ActionIcon
-        onClick={() => (modal.back ? setModal(modal.back) : setModal(undefined))}
-        variant='filled'
-        size='xl'
-        color='yellow'>
-        <ArrowLeft size={50} />
-      </ActionIcon>
-      {modal.type === 'ADD_CAT' && <Text>{t('components.catModal.catModalHeader.addCat')}</Text>}
-      {modal.type === 'VIEW_CAT' && <Text>{t('components.catModal.catModalHeader.viewCat')}</Text>}
-      {modal.type === 'EDIT_CAT' && <Text>{t('components.catModal.catModalHeader.editCat')}</Text>}
+    <>
+      <Group
+        p='md'
+        position='apart'
+        align='center'
+        sx={{
+          width: '100%',
+          borderRadius: '15px',
+        }}>
+        <ActionIcon
+          onClick={() => {
+            if (modal.type === 'VIEW_CAT') {
+              (modal.back ? setModal(modal.back) : setModal(undefined))
```

Comment:
If I understand correctly, the above condition checks if the `modal.back` field is `undefined`. If this is the case, the logic could be reduced to
`setModal(modal.back);`